examples/host_camera_server.cc

```diff
@@ -69,46 +69,39 @@ typedef struct stream_ctx_t
 } stream_ctx_t;
 
 stream_ctx_t *stream_ctx;
-
-char device_index[] = "/dev/video0";
-int width = 640;
-int height = 480;
-int fps = 30;
 AVPacket *pkt;
 
 #define BUF_COUNT 4
 unsigned int buf_count = 0;
 unsigned char *buf_list[BUF_COUNT];
 
-void get_all_dev_nodes()
+std::string get_available_video_device()
 {
-    int fd;
-    struct v4l2_capability video_cap;
-    char dev_name[50];
+    std::string dev_video = NULL;
     for(int devId = 0; devId < 64; devId++) {
-        sprintf(dev_name, "/dev/video%d", devId);
-
-        if((fd = open(dev_name, O_RDONLY)) == -1) {
+        int fd;
+        struct v4l2_capability video_cap = {};
+        std::string dev_name = "/dev/video" + to_string(devId);
+        if((fd = open(dev_name.c_str(), O_RDONLY)) == -1) {
             continue;
-	}
+        }
 
         if(ioctl(fd, VIDIOC_QUERYCAP, &video_cap) == -1)
             cout <<"cam_info: Can't get capabilities\n";
         else {
-	    if(!strcmp((const char*)video_cap.driver, "v4l2 loopback"))
-	        strncpy(device_index, dev_name, 11);
+            if(!strcmp((const char*)video_cap.driver, "v4l2 loopback"))
+                dev_video = dev_name;
```

Comment:
do you actually want to got to the next iteration if you ended up here? Maybe you just want to return from here?

And you can slightly simplify if else if you will use `continue`:
```
        if(ioctl(fd, VIDIOC_QUERYCAP, &video_cap) == -1) {
            cout <<"cam_info: Can't get capabilities\n";
            close(fd);
            continue;
        }
        close(fd);
        if(!strcmp((const char*)video_cap.driver, "v4l2 loopback"))
            return dev_name;
```

Reply:
done